src/app/signup/page.tsx

```diff
@@ -1,14 +1,23 @@
 "use client";
 
-import Image from "next/image";
 import { GradButton } from "@/components/ui/grad-button";
 import SignUpImage from "@/components/ui/signup-image";
+import { SignUpFieldErrors, signupUser } from "@/lib/User/actions";
+import { EMAIL_ALREADY_EXISTS_MSG, signUpSchema } from "@/lib/User/validations";
 import { zodResolver } from "@hookform/resolvers/zod";
+import Image from "next/image";
+import { Fragment, useCallback, useRef, useState } from "react";
 import { useForm } from "react-hook-form";
-import { signupUser } from "@/lib/User/actions";
-import { signUpSchema } from "@/lib/User/validations";
 
 const SignUp = () => {
+  const formRef = useRef<HTMLFormElement>(null);
+  const [serverErrors, setServerErrors] = useState<
```

Comment:
I wonder if we need this. I'll leave it for now and look into refactoring after though.